### DISCUSSION TITLE
HaloAnnotator does not work

### DISCUSSION THREAD

**Discussion by wilsonlv** [roboflow/supervision#1183] [Q&A] [answered]
### Search before asking

- [X] I have searched the Supervision [issues](https://github.com/roboflow/supervision/issues) and found no similar feature requests.

### Question

can you give me a complete code? when i use HaloAnnotator, nothing changes in the image, the detections.mask is None

```python
import cv2
import supervision as sv
from ultralytics import YOLO

filename = '../src/static/dog.png'
image = cv2.imread(filename)

model = YOLO('../models/yolov8x.pt')
results = model(image)[0]

detections = sv.Detections.from_ultralytics(results)

annotated_image = sv.HaloAnnotator().annotate(
    scene=image.copy(), detections=detections)

sv.plot_image(annotated_image)

```

**Comment by LinasKo** [comment, marked as answer]
Hi @wilsonlv :wave:

It's very useful that you checked the `mask` value.

To use `HaloAnnotator` you need a segmentation model. For example:
```python
model = YOLO('yolov8x-seg')
```

**Comment by wilsonlv** [comment]
@LinasKo appreciate for help!

**Reply by SkalskiP** [reply]
@wilsonlv we also mention the `detections.mask` requirement in docs: https://supervision.roboflow.com/latest/detection/annotators/#supervision.annotators.core.HaloAnnotator :)
